#!/bin/bash
#PBS -A UQ-IMB
#PBS -N cutadapt_Lane1
#PBS -l walltime=14:00:00
#PBS -l select=1:ncpus=10:mem=64GB
#PBS -e /30days/s4596423/Algin_output/cut_Lane1_error.txt
#PBS -o /30days/s4596423/Algin_output/cut_Lane1_output.txt

printf "Start Running\n"
#module load anaconda/5.2.0
#source activate /QRISdata/Q1287/Minh_envs/rnaseq_qc/
module load cutadapt/1.18

BASE='/QRISdata/Q1287/Minh_envs/Lane_1_combined/'
OUTPUT='/QRISdata/Q1287/MOAGR_RNASeq/02_STAR_BAM_sabre_demultiplexed_PE/'
GENOMEDIR='/QRISdata/Q1287/Minh_envs/reference_mm10/mm10/'
GTFFILE='/QRISdata/Q1287/Minh_envs/reference_mm10/gencode.vM24.chr_patch_hapl_scaff.annotation.gtf'
ADAPTER='/QRISdata/Q1287/AgingAtlas/RNAseq/qsub_scripts/adapter.fa'
printf "$(cutadapt --version)"
for i in 1 3 4 6 7 8 9 10 12 13 14 15 17 18
do 
    dirname="${BASE}"
    echo "${dirname}"
    read1="$(ls ${dirname}*-${i}_*_R1.fastq)"
   # read2="$(ls ${dirname}*-${i}_*_R2.corrected.fastq)"
   # #echo "${read1}" and "${read2}" and "${OUTPUT}"
    fn1="${read1##*/}"
    fn2="${read2##*/}"
    fn1="${fn1%.fastq}"
    fn2="${fn2%.fastq}"
   # echo "${fn1}" 
   # echo "${fn2}"
   # echo "----------------------"
   # #cat ${read1} | awk 'NR%4==3{$0="+"};1' >> ${dirname}${fn1}.corrected.fastq
   # #cat ${read2} | awk 'NR%4==3{$0="+"};1' >> ${dirname}${fn2}.corrected.fastq
    run_cut1="$(cutadapt -b file:${ADAPTER} --out ${dirname}${fn1}.trimmed.fastq ${read1})"      
   # run_cut2="$(cutadapt -b file:${ADAPTER} --out ${dirname}${fn2}.trimmed.fastq ${read2})"      
    #run_star="$(STAR --runThreadN 12 --runMode alignReads --genomeDir ${GENOMEDIR} --readFilesIn ${read1} ${read2} --outFileNamePrefix ${OUTPUT}${fn1}.03. --outSAMtype BAM SortedByCoordinate --outSAMunmapped Within --outFilterMatchNminOverLread 0.3 --outFilterScoreMinOverLread 0.3 --twopassMode Basic --sjdbGTFfile ${GTFFILE} 75 --quantMode GeneCounts)"

done
printf "\nDone\n"


