#!/bin/bash
#PBS -A UQ-IMB
#PBS -N Alg_Lane1_hiseq
#PBS -l walltime=14:00:00
#PBS -l select=1:ncpus=10:mem=64GB
#PBS -e /30days/s4596423/Algin_output/align_Lane1_hiseq_error_3.txt
#PBS -o /30days/s4596423/Algin_output/align_Lane1_hiseq_output_3.txt

printf "Start Running\n"
module load anaconda/5.2.0
source activate /QRISdata/Q1287/Minh_envs/rnaseq_qc/

BASE='/QRISdata/Q1287/RNAseq/01_demultiplex/Lane_1_samples/'
OUTPUT='/QRISdata/Q1287/RNAseq//02_STAR_BAM_mapping/'
GENOMEDIR='/QRISdata/Q1287/Minh_envs/reference_mm10/mm10/'
GTFFILE='/QRISdata/Q1287/Minh_envs/reference_mm10/gencode.vM24.chr_patch_hapl_scaff.annotation.gtf'

for i in 1 3 4 6 7 8 9 10 12 13 14 15 17 18
do 
    dirname="${BASE}"
    echo "${dirname}"
    read1="$(ls ${dirname}*-${i}_*_R1.trimmed.fastq.gz)"
    #read2="$(ls ${dirname}*-${i}_*_R2.trimmed.fastq)"
    #echo "${read1}" and "${read2}" and "${OUTPUT}"
    fn1="${read1##*/}"
    #fn2="${read2##*/}"
    fn1="${fn1%.fastq.gz}"
    #fn2="${fn2%.fastq}"
    echo "${fn1}" 
    #echo "${fn2}"
    echo '----------------------'
    run_star="$(STAR --runThreadN 12 --runMode alignReads --genomeDir ${GENOMEDIR} --readFilesIn ${read1} --readFilesCommand zcat --outFileNamePrefix ${OUTPUT}${fn1}.03. --outSAMtype BAM SortedByCoordinate --outSAMunmapped Within --outFilterMatchNminOverLread 0.3 --outFilterScoreMinOverLread 0.3 --twopassMode Basic --sjdbGTFfile ${GTFFILE} 75 --quantMode GeneCounts)"

done
printf "\nDone\n"


